### PR TITLE
Temporarily host apt-key.gpg as original url is failing

### PR DESCRIPTION
**Pull request by dims**
NOTE: Once this lands folks will be able to use https://dl.k8s.io/apt/doc/apt-key.gpg instead of the google url. 

A cached version of a gpg key is used from the way back machine, dated 2023-02-23 here:

https://web.archive.org/web/20230223152417/https://packages.cloud.google.com/apt/doc/apt-key.gpg

The live gpg key at https://packages.cloud.google.com/apt/doc/apt-key.gpg is inaccessible with a HTTP 500 response.

When the live gpg key is available agian, we can switch the rewrite rule to point to: https://packages.cloud.google.com/apt/doc/apt-key.gpg and remove the temporary copy from this repository.

NOTE: you can use `gpg --dearmor -o apt-key.gpg apt-key.asc` to generate the binary version from the ascii version. checked in both for easier inspection

Also see other copies of this file: https://cs.github.com/?scopeName=All+repos&scope=&q=xsBNBGKItdQBCADWmKTNZEYWgXy73

### CODE REVIEW

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

File: apps/k8s-io/configmap-nginx.yaml [hunk 1]
```diff
@@ -176,6 +176,7 @@ data:
         listen 80;
 
         location / {
+          rewrite ^/apt/doc/apt-key.gpg    https://raw.githubusercontent.com/kubernetes/k8s.io/main/apt/doc/apt-key.gpg redirect;
```
Comment (aojea) [inline apps/k8s-io/configmap-nginx.yaml:179]: smart

**Comment by Oskang09** [comment]
Hi @dims , it's seems to be still not working yet? 

Commands

```
#!/bin/bash -eo pipefail
sudo apt-get update && sudo apt-get install -y apt-transport-https
curl -4 -s https://dl.k8s.io/apt/doc/apt-key.gpg | sudo apt-key add -
sudo touch /etc/apt/sources.list.d/kubernetes.list
echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee -a /etc/apt/sources.list.d/kubernetes.list
sudo apt-get update
sudo apt-get install -y kubectl
```

Logs

```
Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
Get:2 http://security.debian.org/debian-security bullseye-security InRelease [48.4 kB]
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Get:4 http://security.debian.org/debian-security bullseye-security/main amd64 Packages [229 kB]
Get:5 http://deb.debian.org/debian bullseye/main amd64 Packages [8183 kB]
Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages.diff/Index [16.2 kB]
Ign:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages.diff/Index
Get:7 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [14.6 kB]
Fetched 8651 kB in 1s (8244 kB/s)
Reading package lists... Done
N: Repository 'http://deb.debian.org/debian bullseye InRelease' changed its 'Version' value from '11.1' to '11.6'
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  apt-transport-https
0 upgraded, 1 newly installed, 0 to remove and 77 not upgraded.
Need to get 160 kB of archives.
After this operation, 166 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main amd64 apt-transport-https all 2.2.4 [160 kB]
Fetched 160 kB in 0s (11.9 MB/s)              
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package apt-transport-https.
(Reading database ... 19362 files and directories currently installed.)
Preparing to unpack .../apt-transport-https_2.2.4_all.deb ...
Unpacking apt-transport-https (2.2.4) ...
Setting up apt-transport-https (2.2.4) ...
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
gpg: no valid OpenPGP data found.
```

**Comment by csabato-6river** [comment]
@dims Yes still have issues with the updated url

**Comment by csabato-6river** [comment]
@dims Yes still have issues with the updated url

**Comment by tuxiqae** [comment]
Make sure to add the flag `-L` to your `curl` commands in-order to follow redirections

**Comment by Oskang09** [comment]
> Make sure to add the flag `-L` to your `curl` commands in-order to follow redirections

Added this and working already, thanks a lot 👍

**Comment by dims** [comment]
Folks, if you got here and the new url is work, leave a note either way (good or bad!) if we just see bad things here we can't tell if it worked for you or not and assume the worst! :)

**Comment by jnicholsDLR** [comment]
thanks all! the new url works like a charm.

**Comment by ReggieCarey** [comment]
Please do not forget to update the installation documentation at kubernetes.io.  Is https://dl.k8s.io/apt/doc/apt-key.gpg the final location for this file?

**Comment by stevo-f3** [comment]
Replacement URL works for me too, thanks!

**Comment by sftim** [comment]
If we want to, we can pop a banner up on https://k8s.io/ with a short TTL.

**Comment by StanleyPan** [comment]
I used below two commands, but I'm still getting NO_PUBKEY B53DC80D13EDEF05 error, what am I doing wrong?

`sudo curl -fsSLo /usr/share/keyrings/kubernetes-archive-keyring.gpg https://dl.k8s.io/apt/doc/apt-key.gpg`

`echo "deb [signed-by=/etc/apt/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee /etc/apt/sources.list.d/kubernetes.list`

```
Stanley@stanley-synology-ubuntu-2:~$ sudo apt update
Hit:1 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:2 http://ca.archive.ubuntu.com/ubuntu focal InRelease
Hit:4 http://ca.archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:5 http://ppa.launchpad.net/webupd8team/y-ppa-manager/ubuntu focal InRelease
Hit:6 http://ca.archive.ubuntu.com/ubuntu focal-backports InRelease
Get:3 https://packages.cloud.google.com/apt kubernetes-xenial InRelease [8,993 B]
Err:3 https://packages.cloud.google.com/apt kubernetes-xenial InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
Reading package lists... Done
W: GPG error: https://packages.cloud.google.com/apt kubernetes-xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
E: The repository 'https://apt.kubernetes.io kubernetes-xenial InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.```

**Comment by samgabriel** [comment]
new url works have to remember to delete the old /etc/apt/sources.list.d/kubernetes.list

**Comment by cacods** [comment]
> The new URL https://dl.k8s.io/apt/doc/apt-key.gpg worked for me, the old one still doesn't work

Same here:
AWS EC2 Debian 11

**Comment by Kumm-Kai** [comment]
Since a few days https://packages.cloud.google.com/apt/doc/apt-key.gpg returns the ascii armored GPG key instead of the binary format. It this known to you?

**Comment by Nipun2508** [comment]
it worked

**Comment by zillani** [comment]
> Folks please use https://dl.k8s.io/apt/doc/apt-key.gpg instead of the google url and let us know if you still see issues!

Thanks @dims worked !
```
 sudo curl -fsSLo /usr/share/keyrings/kubernetes-archive-keyring.gpg https://dl.k8s.io/apt/doc/apt-key.gpg
 echo "deb [signed-by=/usr/share/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee /etc/apt/sources.list.d/kubernetes.list
```

**Comment by xmudrii** [comment]
Hello folks,

The legacy package repositories have been deprecated and frozen starting from September 13, 2023. We strongly recommend using the community-owned package repositories hosted at pkgs.k8s.io. See the following announcements for more details:

- https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/
- https://kubernetes.io/blog/2023/08/31/legacy-package-repository-deprecation/

**Comment by jedrekdomanski** [comment]
> pkgs.k8s.io

[pkgs.k8s.io](url) throws Access Denied

**Pull request merged**